### ISSUE TITLE
dat hooks API

### ISSUE THREAD

**Issue by max-mapper** [datproject/dat#139]
@mafintosh started a little down this path with the beforePut and afterGet functions, but I think we should clearly define a hooks API which would enable 3 cool things:
- easier for people to make third party dat modules that hook into stuff
- we can expose the hooks as configuration options in dat.json
- we can use the hooks to capture statistics/throughput data (for powering e.g. datop)
- would allow for a blob-lazy-fetch external module

inspiration https://www.npmjs.org/package/level-hooks

**Comment by onbjerg** [comment]
@maxogden when you say most of it got implemented, how do we do it then? are there any docs or similar?

**Comment by maxogden** [comment]
@onbjerg Hiya this is in the dat alpha era, we are working on a 1.0 release now and things have a changed a bit since the alpha, sorry. One of the things we changed was that we removed the hooks from the beta and haven't put them back in yet. 

We're doing a new version of our graph module now https://github.com/mafintosh/dat-graph and will update the `dat` CLI tool to use it when we're done. Software is hard!
